packages/cms/lib/modules/resource-form-widgets/index.js

```diff
@@ -41,196 +24,240 @@ module.exports = {
+    self.on('apostrophe-docs:afterTrash', 'deleteConfirmationFields');
+    self.on('apostrophe-workflow:afterCommit', 'logCommitData', function (req, data) {
+      self.apos.utils.info('The commit data is', data);
+    });
 
     /** add config **/
     const superLoad = self.load;
 
-    self.load = function(req, widgets, next) {
-        const styles = openstadMap.defaults.styles;
-        const globalData = req.data.global;
+    self.load = function (req, widgets, next) {
+      const styles = openstadMap.defaults.styles;
+      const globalData = req.data.global;
+      widgets.forEach(async (widget) => {
+        const resourceType = widget.resource ? widget.resource : false;
+        const resourceInfo = resourceType ? resourcesSchema.find((resourceInfo) => resourceInfo.value === resourceType) : false;
+        if (!resourceInfo) {
+          return;
+        }
 
-	      widgets.forEach((widget) => {
-            const resourceType = widget.resource ?  widget.resource : false;
-            const resourceInfo = resourceType  ? resourcesSchema.find((resourceInfo) => resourceInfo.value === resourceType) : false;
+        const resourceConfigKey = resourceInfo ? resourceInfo.configKey : false;
+        const resourceConfig = req.data.global.siteConfig && req.data.global.siteConfig[resourceConfigKey] ? req.data.global.siteConfig[resourceConfigKey] : {};
 
-            if (!resourceInfo) {
-              return;
-            }
+        const siteConfig = req.data.global.siteConfig;
+
+        widget.resourceConfig = {
+          titleMinLength: (resourceConfig.titleMinLength) || 10,
+          titleMaxLength: (resourceConfig.titleMaxLength) || 50,
+          summaryMinLength: (resourceConfig.summaryMinLength) || 20,
+          summaryMaxLength: (resourceConfig.summaryMaxLength) || 140,
+          descriptionMinLength: (resourceConfig.descriptionMinLength) || 140,
+          descriptionMaxLength: (resourceConfig.descriptionMaxLength) || 5000,
+        }
 
-            const resourceConfigKey = resourceInfo ? resourceInfo.configKey : false;
-            const resourceConfig = req.data.global.siteConfig && req.data.global.siteConfig[resourceConfigKey] ? req.data.global.siteConfig[resourceConfigKey] : {};
+        widget.resourceEndPoint = resourceInfo.resourceEndPoint;
 
-	          const siteConfig = req.data.global.siteConfig;
+        widget.siteConfig = {
+          openstadMap: {
+            polygon: (siteConfig && siteConfig.openstadMap && siteConfig.openstadMap.polygon) || undefined,
+          },
+        }
+
+        // Add function for rendering raw string with nunjucks templating engine
+        // Yes this ia a powerful but dangerous plugin :), admin only
+        widget.renderString = (rawInput, data) => {
+          try {
+            return self.apos.templates.renderStringForModule(req, rawInput, data, self);
+          } catch (e) {
+            return 'Error....'
```

Comment:
Vergelijkbare code zit ook in de `resource-raw-widgets/index.js`. Hier liep ik een paar keer tegen deze 'Error...' aan, maar dat is erg vervelend debuggen. Kunnen we hier een error loggen? Dat scheelt weer een hoop deployments.